src/clients/go/docs.zig

```diff
@@ -163,7 +163,7 @@ pub const GoDocs = Docs{
 
     .client_object_documentation =
     \\The third argument to `NewClient` is a `uint` max concurrency
-    \\setting. `32` is a good default and can increase to `4096`
+    \\setting. `256` is a good default and can increase to `8190`
```

Comment:
why 8190 (over 8192)?

Reply:
It's from here:

https://github.com/tigerbeetle/tigerbeetle/blob/9f11a15ca2610181a3c3f0b79ebd4315a69de9e6/src/clients/c/tb_client/context.zig#L121-L130

I hate exposing internal constants like this, but we already have this in other places until the client exposes it properly.

Reply:
Would it make sense for the Context-API change to remove the `concurrency_max` limit?

Reply:
@kprotty 
As a second thought, it should not be related to `batch_max`:

- The `concurrency_max` doesn't mean that all concurrent tasks are submitting the same event type!
- We can swap StateMachines (in fact we do it for the EchoTest).

Changed it to `8192` an arbitrary value, just double of the previous one 4096.